tests/func/test_data_status.py

```diff
@@ -423,6 +423,41 @@ def test_missing_remote_cache(M, tmp_dir, dvc, scm, local_remote):
     }
 
 
+@pytest.fixture
+def dvc_pipeline_with_push_false(
+    tmp_dir: TmpDir, dvc: Repo, scm, mocker, local_remote
+) -> None:
+    """DVC pipeline with with two outs, one with push=False."""
+    tmp_dir.gen("fixed", "fixed")
+    dvc.stage.add(
+        name="create-foo", cmd="echo foo > foo", deps=["fixed"], outs_no_push=["foo"]
+    )
```

Comment:
This is a bit verbose, but it's better than adding `outs_no_push` arg to `stage.add()` function.


```suggestion
    stage = dvc.stage.create(
        name="create-foo", cmd="echo foo > foo", deps=["fixed"]
    )
    stage.outs[0].can_push = False
    stage.dump()
```